app/uk/gov/hmrc/plasticpackagingtax/registration/connectors/GeneralPartnershipConnector.scala

```diff
@@ -21,19 +21,22 @@ import play.api.http.Status.CREATED
 import uk.gov.hmrc.http.HttpReads.Implicits._
 import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse, InternalServerException}
 import uk.gov.hmrc.plasticpackagingtax.registration.config.AppConfig
-import uk.gov.hmrc.plasticpackagingtax.registration.connectors.PartnershipConnector.CreateJourneyTimer
+import uk.gov.hmrc.plasticpackagingtax.registration.connectors.GeneralPartnershipConnector.CreateJourneyTimer
 import uk.gov.hmrc.plasticpackagingtax.registration.models.genericregistration.{
-  PartnershipCreateJourneyRequest,
-  PartnershipDetails
+  GeneralPartnershipDetails,
+  PartnershipCreateJourneyRequest
 }
 
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class PartnershipConnector @Inject() (httpClient: HttpClient, config: AppConfig, metrics: Metrics)(
-  implicit ec: ExecutionContext
-) extends GrsIntegration[PartnershipCreateJourneyRequest, PartnershipDetails] {
+class GeneralPartnershipConnector @Inject() (
```

Comment:
@kiranprakashdigital, you might want to consider refactoring this Connector to support all partnership types. Although, when I attempted to support all GRS integrations (in `GrsIntegration`) I failed due to issues with types and json de/serialisation.